src/operator/tensor/ordering_op-inl.h

```diff
@@ -362,7 +362,7 @@ MSHADOW_FORCE_INLINE void TopKSort(const Tensor<gpu, 1, DType>& dat,
     }
   } else {
     const int nthreads(mshadow::cuda::kBaseThreadNum);
-    PartialSortSmallK<<<M, nthreads, nthreads*K*(sizeof(int)+sizeof(DType)),
+    PartialSortSmallK<<<M, nthreads, nthreads*K*(sizeof(IType)+sizeof(DType)),
```

Comment:
Looking at the implementation of this kernel (which does not depend on your IType template), it should always be index_t here.

Reply:
Is it okay if there's a slight memory regression after changing it to index_t? I think @apeforest wanted to decide whether to use int or index_t at runtime.

Reply:
This is not a global memory of the gpu that is used here, it is shared memory amount assigned to this kernel. What is the value of nthreads here?

Reply:
nthreads seems to be a constant of value 256 and is defined in the mshadow/cuda/tensor_gpu-inl.cuh

Reply:
Ok, thanks. So the only real limit here is that we can't request more than 48 kB of shared memory for this kernel, and the change from int to index_t should not matter be problematic here: `256 * (8 + sizeof(DType)) * K <= 256 * (8 + 8) * K = 4096 * K`. K is small (5), so it is nowhere near the limit.

Reply:
Ok great! So just changing the `int` to `index_t` should be okay even though we see a slight memory regression when doing so?

Reply:
Yeah, I don't believe the increase of GPU memory usage that you see comes from this particular change. Shared memory is part of L1 cache of the GPU, it is not counted towards the DRAM size.

Reply:
@ptrendx If we look at the [memory profiling result ](https://github.com/apache/incubator-mxnet/pull/17462#issuecomment-579961169) posted by @JonTanS, there IS a ~10% max memory increase and ~5% average memory increase caused by this solely change. 

@JonTanS could you please re-run the profiler again to make sure the memory result is accurate?

Reply:
@JonTanS could you also move the memory comparison results to the PR description so it's more noticeable to reviewers? Thanks

Reply:
@apeforest sure thing! I'm currently rebuilding mxnet and then will be running the memory usage. I'll move everything to the top once I have the complete results.